### PR TITLE
Remove the forest_vm crate

### LINKED ISSUE #1978: Remove the `forest_vm` crate
**Issue summary**

The `forest_vm` crate contains six source files that can all be refactored out of existence:
- `vm/src/actor_state.rs`: The JSON serialization/de-serialization code should be moved to the `forest_json` crate.
- `vm/src/deal_id.rs`: `DealID` is exported by `fvm_shared`.
- `vm/src/deal_id.rs`: None of this code is used.
- `vm/src/lib.rs`: None of this code is used.
- `vm/src/method.rs`: These re-exports can be refactored out.
- `vm/src/token.rs`: The single re-export can be refactored out.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1978

**Other information and links**

### CODE REVIEW

File: blockchain/blocks/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ forest_fil_types.workspace  = true
 forest_json.workspace       = true
 forest_json_utils.workspace = true
 forest_message              = { workspace = true, features = ["blst"] }
-forest_vm.workspace         = true
+
```
Comment (lemmih) [inline blockchain/blocks/Cargo.toml:22]:
```suggestion
```

File: blockchain/blocks/src/header/mod.rs [hunk 2]
```diff
@@ -128,7 +127,7 @@ pub struct BlockHeader {
     bls_aggregate: Option<Signature>,
     /// the base fee of the parent block
     #[builder(default)]
-    parent_base_fee: TokenAmount,
+    parent_base_fee: BigInt,
```
Comment (lemmih) [inline blockchain/blocks/src/header/mod.rs:130]:
```suggestion
    parent_base_fee: TokenAmount,
```

File: blockchain/state_manager/src/lib.rs [hunk 3]
```diff
@@ -1401,7 +1400,7 @@ where
         self: &Arc<Self>,
         height: ChainEpoch,
         state_tree: &StateTree<&DB>,
-    ) -> Result<TokenAmount, anyhow::Error> {
+    ) -> Result<BigInt, anyhow::Error> {
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:1403]:
```suggestion
    ) -> Result<TokenAmount, anyhow::Error> {
```

File: vm/message/src/lib.rs [hunk 4]
```diff
@@ -22,25 +23,25 @@ pub trait Message {
     /// Returns the message sequence or nonce.
     fn sequence(&self) -> u64;
     /// Returns the amount sent in message.
-    fn value(&self) -> &TokenAmount;
+    fn value(&self) -> &BigInt;
     /// Returns the method number to be called.
-    fn method_num(&self) -> MethodNum;
+    fn method_num(&self) -> u64;
```
Comment (lemmih) [inline vm/message/src/lib.rs:28]:
```suggestion
    fn method_num(&self) -> MethodNum;
```

**Review by lemmih — CHANGES_REQUESTED** [review]
We still want to use `TokenAmount` and `MethodNum`. Just import them directly from `fvm` and `fvm_shared`.

File: vm/message/src/lib.rs (hunk 4 above)
Comment (tyshko5) [reply vm/message/src/lib.rs]: Fixed

File: blockchain/state_manager/src/lib.rs (hunk 3 above)
Comment (tyshko5) [reply blockchain/state_manager/src/lib.rs]: Fixed

File: blockchain/blocks/src/header/mod.rs (hunk 2 above)
Comment (tyshko5) [reply blockchain/blocks/src/header/mod.rs]: Fixed

File: blockchain/blocks/Cargo.toml (hunk 1 above)
Comment (tyshko5) [reply blockchain/blocks/Cargo.toml:22]: Fixed

**Review by lemmih — CHANGES_REQUESTED** [review]
There are a bunch more times you've used `BigInt` instead of `TokenAmount`. They all need to be fixed.

File: blockchain/blocks/Cargo.toml (hunk 1 above)
Comment (lemmih) [reply blockchain/blocks/Cargo.toml:22]: As of writing, this has not been fixed. Please make sure you always push all your patches before commenting.

File: blockchain/state_manager/src/vm_circ_supply.rs [hunk 5]
```diff
@@ -202,16 +202,14 @@ fn get_fil_power_locked<DB: BlockStore>(
 fn get_fil_reserve_disbursed<DB: BlockStore>(
     state_tree: &StateTree<DB>,
 ) -> Result<TokenAmount, anyhow::Error> {
-    let fil_reserved: BigInt = BigInt::from(300_000_000) * FILECOIN_PRECISION;
+    let fil_reserved: TokenAmount = TokenAmount::from(300_000_000) * FILECOIN_PRECISION;
```
Comment (lemmih) [inline blockchain/state_manager/src/vm_circ_supply.rs:205]: Try to keep `TokenAmount` being a consistent unit:
```suggestion
    let fil_reserved: TokenAmount = BigInt::from(300_000_000) * FILECOIN_PRECISION;
```

File: blockchain/blocks/Cargo.toml (hunk 1 above)
Comment (lemmih) [inline blockchain/blocks/Cargo.toml:22]: Still an issue:

```suggestion
```

File: node/rpc-api/Cargo.toml [hunk 6]
```diff
@@ -21,7 +21,7 @@ forest_libp2p.workspace          = true
 forest_message                   = { workspace = true, features = ["blst"] }
 forest_message_pool.workspace    = true
 forest_state_manager.workspace   = true
-forest_vm.workspace              = true
+
```
Comment (lemmih) [inline node/rpc-api/Cargo.toml:24]: You sure you want a newline here?
```suggestion
```

File: types/Cargo.toml [hunk 7]
```diff
@@ -16,10 +16,9 @@ base64.workspace = true
 cid              = { workspace = true, default-features = false, features = ["std"] }
 # cannot update due to `blst` version conflict
 filecoin-proofs-api         = { version = "11.0", default_features = false, optional = true }
-forest_encoding.workspace   = true
 forest_json.workspace       = true
 forest_json_utils.workspace = true
-forest_vm.workspace         = true
+
```
Comment (lemmih) [inline types/Cargo.toml:21]:
```suggestion
```

File: vm/message/Cargo.toml [hunk 8]
```diff
@@ -18,7 +18,7 @@ forest_crypto               = { workspace = true, default-features = false }
 forest_encoding.workspace   = true
 forest_json.workspace       = true
 forest_json_utils.workspace = true
-forest_vm.workspace         = true
+
```
Comment (lemmih) [inline vm/message/Cargo.toml:21]:
```suggestion
```

**Review by lemmih — CHANGES_REQUESTED** [review]

File: vm/actor_interface/src/builtin/market/mod.rs [hunk 9]
```diff
@@ -87,7 +86,7 @@ impl State {
     }
 
     /// Consume state to return just total funds locked
-    pub fn total_locked(&self) -> TokenAmount {
+    pub fn total_locked(&self) -> BigInt {
```
Comment (lemmih) [inline vm/actor_interface/src/builtin/market/mod.rs:89]:
```suggestion
    pub fn total_locked(&self) -> TokenAmount {
```

File: vm/actor_interface/src/builtin/market/mod.rs [hunk 10]
```diff
@@ -156,11 +155,11 @@ pub struct DealProposal {
     pub start_epoch: ChainEpoch,
     pub end_epoch: ChainEpoch,
     #[serde(with = "json")]
-    pub storage_price_per_epoch: TokenAmount,
+    pub storage_price_per_epoch: BigInt,
```
Comment (lemmih) [inline vm/actor_interface/src/builtin/market/mod.rs:158]:
```suggestion
    pub storage_price_per_epoch: TokenAmount,
```

File: vm/actor_interface/src/builtin/market/mod.rs [hunk 11]
```diff
@@ -156,11 +155,11 @@ pub struct DealProposal {
...
     #[serde(with = "json")]
-    pub provider_collateral: TokenAmount,
+    pub provider_collateral: BigInt,
```
Comment (lemmih) [inline vm/actor_interface/src/builtin/market/mod.rs:160]:
```suggestion
    pub provider_collateral: TokenAmount,
```

File: vm/actor_interface/src/builtin/market/mod.rs [hunk 12]
```diff
@@ -156,11 +155,11 @@ pub struct DealProposal {
...
     #[serde(with = "json")]
-    pub client_collateral: TokenAmount,
+    pub client_collateral: BigInt,
```
Comment (lemmih) [inline vm/actor_interface/src/builtin/market/mod.rs:162]:
```suggestion
    pub client_collateral: TokenAmount,
```

**Review by lemmih — CHANGES_REQUESTED** [review]
I've marked a few more issues. There are many more. Make sure to go through all of your changes here on GitHub.

File: blockchain/state_manager/src/vm_circ_supply.rs (hunk 5 above)
Comment (tyshko5) [reply blockchain/state_manager/src/vm_circ_supply.rs]: Returned as is was before.

**Review by lemmih — APPROVED** [review]
Good. One less crate to worry about.

**Review by elmattic — APPROVED** [review]

**Pull request merged**